awscrt/s3.py

```diff
@@ -517,13 +545,18 @@ def _on_finish(self, error_code, status_code, error_headers, error_body):
         error = None
         if error_code:
             error = awscrt.exceptions.from_code(error_code)
-            if error_body:
-                # TODO The error body is XML, will need to parse it to something prettier.
-                try:
-                    extra_message = ". Body from error request is: " + str(error_body)
-                    error.message = error.message + extra_message
-                except BaseException:
-                    pass
+
+            # Make this into an S3ResponseError, if possible
+            if isinstance(error, awscrt.exceptions.AwsCrtError) \
+                    and status_code is not None \
+                    and error_headers is not None:
+                error = S3ResponseError(
+                    code=error.code,
+                    name=error.name,
+                    message=error.message,
+                    status_code=status_code,
+                    headers=error_headers,
+                    body=error_body)
             self._finished_future.set_exception(error)
```

Comment:
How about always throwing an `S3ResponseError`, which will contain as much information as we have, instead of two different exceptions?

Reply:
This is also helpful for cases when we add cause to `S3ResponseError`

Reply:
The idea is to only raise S3ResponseError with the extra data when we actually have the extra data